eng/scripts/Language-Settings.ps1

```diff
@@ -348,7 +369,45 @@ function Get-DocsCiLine ($item) {
   return $line
 }
 
+function EnsureCustomSource($package) {
+  # $PackageSourceOverride is a global variable provided in 
+  # Update-DocsMsPackages.ps1. Its value can set a "customSource" property.
+  # If it is empty then the property is not overridden.
+  if (!$PackageSourceOverride) {
+    return $package
+  }
+
+  Write-Host "Checking custom package source for $($package.Name)"
+  $existingVersions = Find-Package `
+    -Name $package.Name `
+    -Source CustomPackageSource `
+    -AllVersions `
+    -AllowPrereleaseVersions
+  
+  # Matches package version against output: 
+  # "Azure.Security.KeyVault.Secrets 4.3.0-alpha.20210915.3"
+  $matchedVersion = $existingVersions.Where({$_.Version -eq $package.Versions})
+
+  if (!$matchedVersion) { 
+    return $package
+  }
+
+  $package.Properties['customSource'] = $PackageSourceOverride
+  return $package
+}
+
 function Update-dotnet-DocsMsPackages($DocsRepoLocation, $DocsMetadata) {
+  # $PackageSourceOverride is a global variable provided in 
+  # Update-DocsMsPackages.ps1.
+  if ($PackageSourceOverride) {
```

Comment:
Similarly you will need to dynamically access or check to see if this variable exists.